### PR TITLE
Timetable made to change automatically depending on current Semester

### PR DESCRIPTION
**Pull request by pratik0197**
The Timetable was fixed to enable automatic changing of timetable of first year students with change in semester.
Mathematics-I and Physics-I also change to their -II depending upon the current semester.

Demonstration of changes done, so as to automatically change the schedule of sections based on the current semester
demo2

### CODE REVIEW

File: scp/lib/time_table.dart [hunk 1]
```diff
@@ -26,38 +25,46 @@ class TimeTableState extends State<TimeTable> {
   final double unitHeight = 80.0;
   double screenWidth, screenHeight;
 
+
   Future _fetchSectionData(BuildContext context) async {
+    TimeTableResources.setMathsPhysicsCourseNumber(); // Sets the course number of Maths and Physics for corresponding semester.
```
Comment (DesignrKnight) [inline scp/lib/time_table.dart:30]: Rename to setCourseNumber. We want it to be scalable if new subjects come

File: scp/lib/time_table.dart [hunk 2]
```diff
@@ -26,38 +25,46 @@ class TimeTableState extends State<TimeTable> {
...
-        (theorySection.compareTo('B') == 0)) {
-      sectionSequence = 'tp';
+    if((theorySection.compareTo('Ar.')==0)||(theorySection.compareTo('A')==0)||(theorySection.compareTo('D')==0)||(theorySection.compareTo('C')==0)||(theorySection.compareTo('B')==0))
+      {
+        sectionSequence = currentMonth>=8 ? 'tp' : 'pt';
+      }else{
+      sectionSequence = currentMonth>=8 ? 'pt' : 'tp';
+    }
+
+    print("Sequence"+sectionSequence);
```
Comment (DesignrKnight) [inline scp/lib/time_table.dart:42]: remove print statements

**Comment by DesignrKnight** [comment]
Are you using new routes done as per our master?
Check!

**Comment by DesignrKnight** [comment]
@ankank30 
Please review when free

File: scp/lib/time_table.dart (hunk 2 above)
Comment (pratik0197) [reply scp/lib/time_table.dart:42]: Yes yes.

**Comment by pratik0197** [comment]
> Are you using new routes done as per our master?
> Check!

Yes. I have not manipulated any routes. :)

**Review by DesignrKnight — COMMENTED** [review]
> > Are you using new routes done as per our master?
> > Check!
> 
> Yes. I have not manipulated any routes. :)

Our master was updated a few days ago. That's why

File: scp/lib/userdata.dart [hunk 3]
```diff
@@ -91,7 +91,7 @@ class UserdataState extends State<Userdata> {
                                     hintText: 'Enter Roll No.',
                                     border: InputBorder.none),
                                 onChanged: (value) {
-                                  rollNo = value;
+                                  rollNo = value.trim();
                                 },
                                 controller: rollController,
                               ),
```
[the remark this answers is not in the document]
Comment (pratik0197) [reply scp/lib/userdata.dart:97]: Yes. Removing this.

File: scp/lib/time_table.dart [hunk 4]
```diff
@@ -26,38 +25,46 @@ class TimeTableState extends State<TimeTable> {
...
+    print("Theory"+theorySection);
+    print("Practical"+practicalSection);
+    if(currentMonth < 8){
```
[the remark this answers is not in the document]
Comment (pratik0197) [reply scp/lib/time_table.dart:42]: Cool

File: scp/lib/time_table.dart [hunk 5]
```diff
@@ -26,38 +25,46 @@ class TimeTableState extends State<TimeTable> {
...
-        (theorySection.compareTo('D') == 0) ||
-        (theorySection.compareTo('C') == 0) ||
-        (theorySection.compareTo('B') == 0)) {
-      sectionSequence = 'tp';
+    if((theorySection.compareTo('Ar.')==0)||(theorySection.compareTo('A')==0)||(theorySection.compareTo('D')==0)||(theorySection.compareTo('C')==0)||(theorySection.compareTo('B')==0))
+      {
+        sectionSequence = currentMonth>=8 ? 'tp' : 'pt';
+      }else{
+      sectionSequence = currentMonth>=8 ? 'pt' : 'tp';
+    }
```
[the remark this answers is not in the document]
Comment (pratik0197) [reply scp/lib/time_table.dart:40]: Yes. I'm on it make it less redundant

File: scp/lib/time_table.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (pratik0197) [reply scp/lib/time_table.dart:42]: > > > Are you using new routes done as per our master?
> > > Check!
> > 
> > 
> > Yes. I have not manipulated any routes. :)
> 
> Our master was updated a few days ago. That's why

Yes.. I merged the upstream master today only.

File: scp/lib/time_table.dart [hunk 6]
```diff
@@ -26,38 +25,42 @@ class TimeTableState extends State<TimeTable> {
...
+  _resetSections(BuildContext context) async{
+    SharedPreferences sharedPreferences=await SharedPreferences.getInstance();
     sharedPreferences.getKeys();
     sharedPreferences.remove('theory_section');
     sharedPreferences.remove('prac_section');
     sharedPreferences.setBool('show_timetable', false);
-    Navigator.of(context).pushNamedAndRemoveUntil(
-        Routes.rHomepage, (Route<dynamic> route) => false);
+    Navigator.of(context)
+        .pushNamedAndRemoveUntil('/homePage', (Route<dynamic> route) => false);
```
Comment (DesignrKnight) [inline scp/lib/time_table.dart:55]: The removed line here is the actual route in upstream

Comment (pratik0197) [reply scp/lib/time_table.dart:55]: Okay okay. have changed this.

**Pull request closed without merging**